### PR TITLE
URL-encode/decode resource names for HTTP API part 2

### PR DESCRIPTION
**Pull request by dttung2905**
### Overview
This MR solves some parts of the issues in #11258. I intend to break it down into several smaller MRs to review those easily

Thanks @littlestar642 for his work . With #11335, we have with a generic function `getPathSuffixUnescaped` so I just apply it where necessary

### CODE REVIEW

**Comment by dttung2905** [comment]
@jkirschner-hashicorp Could you help taking a look at it ?

**Comment by dttung2905** [comment]
Hi @kisunji, do you mind reviewing this MR?

File: agent/catalog_endpoint.go [hunk 1]
```diff
@@ -2,11 +2,9 @@ package agent
 
 import (
 	"fmt"
-	"net/http"
-	"strings"
-
 	metrics "github.com/armon/go-metrics"
 	"github.com/armon/go-metrics/prometheus"
+	"net/http"
```
Comment (kisunji) [inline agent/catalog_endpoint.go:7]: Could you group this with `"fmt"` above? We prefer:
```
stdlib

third party

source package
```

File: agent/coordinate_endpoint.go [hunk 2]
```diff
@@ -2,11 +2,9 @@ package agent
 
 import (
 	"fmt"
+	"github.com/hashicorp/consul/agent/structs"
```
Comment (kisunji) [inline agent/coordinate_endpoint.go:5]: Same as above; please move this import into its own grouping below

**Review by kisunji — COMMENTED** [review]
Looks good! Just a minor nit-pick comment with imports, but once that's fixed this is good to merge 👍

**Comment by dttung2905** [comment]
@kisunji all changes have been made like you suggested

**Review by kisunji — APPROVED** [review]
Thanks for your contribution!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/547464.

**Comment by jkirschner-hashicorp** [comment]
@dttung2905, @kisunji : I updated the list in the issue description accordingly: https://github.com/hashicorp/consul/issues/11258#issue-1021341001

**Comment by kisunji** [comment]
I believe this will need a follow up PR encoding the URLs in the `api` package:
For example, these lines should URLEncode the args before concatenating https://github.com/hashicorp/consul/blob/29f192a130ef473ba84881bc52338b71c04453e9/api/catalog.go#L257-L259

**Comment by dttung2905** [comment]
@kisunji oops. Let me make a separte MR to address this URLEncode issue. Thanks for pointing it out

**Pull request merged**